src/OneDriveSdk.WinRT/Authentication/AdalServiceInfoProvider.cs

```diff
@@ -65,12 +64,14 @@ public AdalServiceInfoProvider(IAuthenticationProvider authenticationProvider, I
 
             var serviceInfo = await base.GetServiceInfo(appConfig, null, httpProvider, clientType);
 
-            serviceInfo.BaseUrl = appConfig.ActiveDirectoryServiceEndpointUrl;
             serviceInfo.ServiceResource = appConfig.ActiveDirectoryServiceResource;
 
-            if (string.IsNullOrEmpty(serviceInfo.ReturnUrl))
+            if (string.IsNullOrEmpty(serviceInfo.BaseUrl) && !string.IsNullOrEmpty(serviceInfo.ServiceResource))
             {
-                serviceInfo.ReturnUrl = WebAuthenticationBroker.GetCurrentApplicationCallbackUri().ToString();
+                serviceInfo.BaseUrl = string.Format(
+                    Constants.Authentication.OneDriveBusinessBaseUrlFormatString,
+                    serviceInfo.ServiceResource.TrimEnd('/'),
```

Comment:
No, service resource is actually supposed to have the trailing slash. But I didn't want to take that dependency in case there's a point where it's not needed.